### PR TITLE
Add rudimentary language filters

### PR DESCRIPTION
**Pull request by g-k**
refs: https://github.com/servo/servo-starters/issues/8
- Assumes all repos implicitly have a rust tag (this is wrong for highfive and possibly other projects) 
- Defaults to having all languages selected
- Clicking a language tag includes or excludes issues with that tag 

Code might need some cleanup (normalized indentation, merge labels and filter labels, maybe get rid of some for loops).

screen shot 2015-12-01 at 1 16 00 am

### CODE REVIEW

**Comment by jdm** [comment]
Fantastic! I'll take a look at this later today :D

**Comment by jdm** [comment]
Also the JS linter reports:

```
site.js: line 39, col 6, Unnecessary semicolon.
site.js: line 375, col 81, Mixed double and single quotes.
```

File: site.js [hunk 1]
```diff
@@ -194,14 +269,32 @@ var IssueList = React.createClass({
         };
     },
 
+    getIssuesWithSelectedLabels: function (issues, selectedLabels) {
+        var labelNames = selectedLabels.map(function (label) {
+            return label.name;
+        });
+
+        var filteredIssues = issues.filter(function (issue) {
+        for (var i = 0; i < issue.labels.length; i++) {
```
Comment (jdm) [inline site.js]: This line should be indented.

**Comment by jdm** [comment]
This looks great! I have one change I'd like made, though: let's add `var repoDefaults = [{ repo: "highfive", language: "L-python" }, { repo: "servo-starters", language: "L-javascript" }]`. We can then change `addDefaultRustLabel` to `addDefaultLanguageLabel` and make it check the array for a language, and default to Rust if there's no matching entry. Does that make sense?

**Comment by g-k** [comment]
Repo default languages seem to work.

Should it be "I want to work with ..." or "I want to work with:"?

I fiddled with the styles too:

screen shot 2015-12-03 at 8 06 52 pm

**Comment by jdm** [comment]
Looks great! Squash these into one commit and I'll merge it :)

**Pull request merged**